app/src/main/java/com/epfl/beatlink/viewmodel/map/user/MapUsersViewModel.kt

```diff
@@ -94,13 +94,23 @@ open class MapUsersViewModel(private val repository: MapUserRepository) : ViewMo
   }
 
   /** Delete a user from the database. */
-  fun deleteMapUser() {
-    viewModelScope.launch {
+  fun deleteMapUser(): Boolean {
+    var result = false
+    return try {
       repository.deleteMapUser(
-          onSuccess = { /* Optionally refresh users or perform other actions */},
-          onFailure = { /* Handle any failures as needed */})
-      _mapUser.value = null
-      _playbackState.value = null
+          onSuccess = {
+            _mapUser.value = null
+            _playbackState.value = null
+            result = true
+          },
+          onFailure = {
+            Log.e("MapUserViewModel", "Failed to delete map user")
+            result = false
+          })
+      result
```

Comment:
the return of the result variable is not needed, just return directly true or false onSuccess or onFailure